### PR TITLE
Update documentation on contributing

### PR DESCRIPTION
**Pull request by kataqatsi**
As i was following along the docs,
This command builds successfully:
```terminal
cd redwood/
yarn run build:test-project ~/my-repos/redwood-project --typescript --link
```

but then errors out when trying to run the next example snippet:
```terminal
cd redwood-project
RWFW_PATH=~/redwood yarn rwfw project:sync
```

passing a different parameter to build:test-project  worked
```diff
-yarn run build:test-project ~/my-repos/redwood-project --typescript --canary
+yarn run build:test-project ~/my-repos/redwood-project --typescript --link
```

### CODE REVIEW

**Comment by thedavidprice** [comment]
Hi @kataqatsi Thanks for opening this PR!

I think the error here is in the paths:
- `RWFW_PATH=~/redwood` --> from the looks of it, this should be `RWFW_PATH=~/my-repos/redwood`
- since the project is being built at `~/my-repos/redwood-project`

--link and --canary just specify the package versions and/or whether or not to use the local packages. Whichever you use will not affect `yarn rwfw project:sync`, which will effectively overwrite existing.

**Comment by kataqatsi** [comment]
Oh sorry, I meant even with `RWFW_PATH` pointing to the path of this cloned repo.

If I build the test project with the `--link` flag when using `build:test-project` everything works (took me a while to figure this out but yay!)

But for some reason when I pass the `--canary` flag  OR no extra flags to `build:test-project`, the test-project will be created without errors, but then the step to later run `project:sync` from the newly created test project errors out with this
```
Found 2 errors.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Command failed with exit code 2: yarn build:types
    at makeError (/home/kataqatsi/Dev/redwoodjs/redwood/node_modules/execa/lib/error.js:60:11)
    at Function.module.exports.sync (/home/kataqatsi/Dev/redwoodjs/redwood/node_modules/execa/index.js:194:17)
    at buildPackages (file:///home/kataqatsi/Dev/redwoodjs/redwood/tasks/framework-tools/src/lib/framework.mjs:178:9)
    at FSWatcher.<anonymous> (file:///home/kataqatsi/Dev/redwoodjs/redwood/tasks/framework-tools/src/frameworkSyncToProject.mjs:110:7)
    at FSWatcher.emit (events.js:400:28)
    at FSWatcher.emitWithAll (/home/kataqatsi/Dev/redwoodjs/redwood/node_modules/chokidar/index.js:541:32)
    at awfEmit (/home/kataqatsi/Dev/redwoodjs/redwood/node_modules/chokidar/index.js:607:14)
    at /home/kataqatsi/Dev/redwoodjs/redwood/node_modules/chokidar/index.js:728:9
    at FSReqCallback.oncomplete (fs.js:193:5) {
  shortMessage: 'Command failed with exit code 2: yarn build:types',
  command: 'yarn build:types',
  escapedCommand: '"yarn build:types"',
  exitCode: 2,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}

rwfw  Error building @redwoodjs/codemods...
```

It looks like [someone else was experiencing this as well](https://discord.com/channels/679514959968993311/747258086569541703/913228401903558687)

the typescript errors:
```
packages/api/src/validations/validations.ts:567:32 - error TS2345: Argument of type '(tx: PrismaClient) => Promise<any>' is not assignable to parameter of type 'PrismaPromise<any>[]'.
  Type '(tx: PrismaClient<PrismaClientOptions, never, RejectOnNotFound | RejectPerOperation | undefined>) => Promise<...>' is missing the following properties from type 'PrismaPromise<any>[]': pop, push, concat, join, and 28 more.

567   return await db.$transaction(async (tx: PrismaClient) => {
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

packages/api/src/validations/validations.ts:568:25 - error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'PrismaClient<PrismaClientOptions, never, RejectOnNotFound | RejectPerOperation | undefined>'.
  No index signature with a parameter of type 'string' was found on type 'PrismaClient<PrismaClientOptions, never, RejectOnNotFound | RejectPerOperation | undefined>'.

568     const found = await tx[model].findFirst({ where })
```

Running `build:clean` before creating the test project didn't do anything either.

Only running `build:test-project` with the `--link` flag seemed to successfully get the `project:sync` command working for me.

The .gitpod.yml is [running it like that too](https://github.com/redwoodjs/redwood/blob/ae08fe06c825761c0d4e6e304dcf0fbb64d9b6cb/.gitpod.yml#L26)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -72,17 +72,20 @@ But how do you actually test your changes in the Redwood Framework in your Redwo
...
-> | `--help `             | Show help                                                          |
+> | Arguments & Options   | Description                                                                           |
+> |-----------------------|---------------------------------------------------------------------------------------|
+> | `<project directory>` | Directory to build test project [default: "./blog-test-project"]                      |
+> | `--typescript, --ts`  | Generate a TypeScript project, JavaScript by default [default: JavaScript]            |
+> | `--link`              | Link the current checked out branch of the framework in the project [default: false]  |
+> | `--verbose`           | Verbose output [default: false]                                                       |
+> | `--clean`             | Delete existing directory and recreate project [default: false]                       |
+> | `--canary`            | Upgrade project to latest canary version, NOT compatible with --link [default: true]  |
+> | `--help `             | Show help                                                                             |
```
Comment (thedavidprice) [inline CONTRIBUTING.md:83]:
```suggestion
> | Arguments & Options   | Description                                                                           |
> |-----------------------|---------------------------------------------------------------------------------------|
> | `<project directory>` | Directory to build test project [default: "./blog-test-project"] |
> | `--typescript, --ts`  | Generate a TypeScript project [default: javascript] |
> | `--link`              | Copy Framework dependencies and packages into Test-project [default: false]  |
> | `--verbose`           | Verbose output [default: false]                                                       |
> | `--clean`             | Delete existing directory and recreate Test-project [default: false] |
> | `--canary`            | Upgrade project to latest canary version; NOT compatible with --link [default: true]  |
> | `--help `             | Show help                                                                             |
```

**Comment by thedavidprice** [comment]
Thank you @kataqatsi I'm following you now!

I made some editorial changes. Getting this merged now 🚀

**Pull request merged**